p2p/http/libp2phttp.go

```diff
@@ -406,11 +406,14 @@ type PeerMetadataGetter interface {
 }
 
 type streamRoundTripper struct {
-	server      peer.ID
-	addrsAdded  sync.Once
-	serverAddrs []ma.Multiaddr
-	h           host.Host
-	httpHost    *Host
+	server peer.ID
+	// if true, we won't add the server's addresses to the peerstore. This
+	// should only be set when creating the struct.
+	skipAddAddrs bool
```

Comment:
Can you elaborate on why we need this?

Reply:
It really doesn't matter, but it saves us the synchronization overhead of the `addrsAdded sync.Once` call.